examples/src/keyring/aws_kms/discovery_decrypt_with_preferred_regions.py

```diff
@@ -0,0 +1,109 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""
+When you give the KMS keyring specific key IDs it will use those CMKs and nothing else.
+This is true both on encrypt and on decrypt.
+However, sometimes you need more flexibility on decrypt,
+especially if you might not know beforehand which CMK was used to encrypt a message.
+To address this need, the KMS keyring also supports "discovery" mode.
+In discovery mode, the KMS keyring will do nothing on encrypt
+but will attempt to decrypt *any* data keys that were encrypted under a KMS CMK.
+
+However, sometimes you need to be a *bit* more restrictive than that.
+To address this need, you can use a client supplier to restrict what regions a KMS keyring can talk to.
+
+A more complex but more common use-case is that you would *prefer* to stay within a region,
+but you would rather make calls to other regions than fail to decrypt the message.
+In this case, you want a keyring that will try to decrypt data keys in this region first,
+then try other regions.
```

Comment:
How do we ensure this behavior? Is our multikeyring always ordered? Might be worth noting in the example.

Kind of related to this, should this instead be a multikeyring example?

Reply:
> How do we ensure this behavior? Is our multikeyring always ordered? Might be worth noting in the example.

I have a comment explaining that at the point in the example where we create the multikeyring.


> Kind of related to this, should this instead be a multikeyring example?

IMO no because we're demonstrating a special way to interact with KMS that just happens in involve using the multi-keyring.